cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -136,7 +136,9 @@ private Runnable createCleanupTask(final Object... resources) {
     return new Runnable() {
       @Override
       public void run() {
-        for (Object resource : resources) {
+        List<Object> resourceList = ImmutableList.copyOf(resources);
+        Collections.reverse(resourceList);
```

Comment:
In general, resource cleanup should in reverse order of creation. In this particular case, we close the classloader before deleting the directory